README.md

```diff
@@ -1,2 +1,126 @@
+
+To extend the documentation you can use a `docs` directory or a `README.md` file.
+
+### Use a docs directory
+
+Make a `docs` directory in the root directory of your repository and add markdown files to it. To use add your new files to the docs you will need to copy the file [./.github/actions/mkdocs-pages/mkdocs.yml](./.github/actions/mkdocs-pages/mkdocs.yml) to your repository in a `.github/actions/mkdocs` directory. From here add to the `nav` panel to add your new docs.
+
+**Example**
+
+ Given a folder structure like this add markdown files to the `docs`
+```
+ ├── .github/
+ │   ├── mkdocs/
+ │   │         └── mkdocs.yml
+ │   └── workflows/publish-docs.yml
+ ├── docs/
+         └──Example.md
+```
+ Once you have added your documentation you need to add it to the navigation on the `.github/mkdocs/mkdocs.yml`. Open the file and add your page in the nav section:
+ ```yaml
+ INHERIT: mkdocs-default.yml
+ nav:
+   - index.md
+   - Example.md
+   - About: about/index.md
+ ```
+
+### Using nested README.md
+
+If you want to use a nested `README.md` structure for your documentation create the following `mkdocs.yml` in `.github/mkdocs/mkdocs.yml`.
+```yaml
+INHERIT: mkdocs-nested.yml
+```
+The nested implementation uses the package `awesome-pages` which collapses down folders with single files to a single web page.
+
+## Using dedicated mkdocs settings
+
+You can override any of the settings from your repo, by placing configurations in the `./github/mkdocs` folder in your repo and documentation in your `docs` folder. For example, if your repository has a `./github/mkocs/mkdocs.yml` this will take precedence over [./.github/actions/mkdocs-pages/mkdocs.yml](./.github/actions/mkdocs-pages/mkdocs.yml). 
+
+Please note that adding new files in `docs` requires changing the `nav` in `./github/mkocs/mkdocs.yml`. When you add to `docs` you will require `./github/mkocs/mkdocs.yml` to add these pages to the navigation see [here](#use-a-docs-directory).
+
+## Merging defaults and dedicated settings
+
+To ease the use of defautls and override the required portions only, the local `./github/mkocs/mkdocs.yml` can use the INHERIT mechanism defined by mkdocs and inherit defaults from the [./.github/actions/mkdocs-pages/mkdocs-default.yml](./.github/actions/mkdocs-pages/mkdocs-default.yml), like this one [./.github/actions/mkdocs-pages/mkdocs.yml](./.github/actions/mkdocs-pages/mkdocs.yml).
+
+For example, to change the site name, the repo using the action should have a file `./.github/mkdocs/mkdocs.yml` with content:
+
+```yaml
+INHERIT: mkdocs-default.yml
+
+site_name: my-site-name
+```
+
+If your pages setup requires additional Python packages, you can specify them in [.github/mkdocs/requirements.txt](.github/mkdocs/requirements.txt) The action will install these dependencies automatically.
```

Comment:
the requirements.txt will replace the one in the action, so maybe it's worth adding a note that this does not `add` but rather `replaces`